process/block/metablock.go

```diff
@@ -119,6 +119,12 @@ func NewMetaProcessor(arguments ArgMetaProcessor) (*metaProcessor, error) {
 	return &mp, nil
 }
 
+func (mp *metaProcessor) setEpochForRequestHandler(headerHandler data.HeaderHandler) {
+	if headerHandler != nil {
```

Comment:
move the request a little later than after it is check of nil, than you can set the request without this check.

Reply:
done